### DISCUSSION TITLE
Loading a GeoTIFF is slow on Android (and maybe iOS)

### DISCUSSION THREAD

**Discussion by jameslavery** [Mapsui/Mapsui#2214] [Q&A]
I'm loading a GeoTIFF which is about 2.7MB

My call to create the new GeoTiffProvider is taking 10s of seconds:

`var gif = new GeoTiffProvider(examplePath, new List<Color> { Color.Red });`

Once it's loaded, peformance is good.

Is there something I can do to diagnose why it's so slow, or something I can do with the TIFF format to improve this?

This is currently on an Android emulator, running on a Mac. I appreciate that performance is very likely to be better on a physical device.

**Comment by inforithmics** [comment]
Currently the tiff image is converted pixel by pixel to a skbitmap and the copied again to MRasterFeature. I think this can certainly be improved less copying and faster convertion

**Reply by jameslavery** [reply]
That sounds like an interesting improvement. I'll take a look - though I may need some pointers on where to look!

**Reply by jameslavery** [reply]
Or is it possible to persist the resulting MRasterFeature once it's processed, so that it doesn't have to be re-processed next time?

**Reply by inforithmics** [reply]
https://stackoverflow.com/questions/50312937/skiasharp-tiff-support

I found a stackoverflow question for I think using this would optimize things

**Reply by inforithmics** [reply]
You could always save mrasterfeature images byte array to the file system and load it from there instead of processing the tiff file

**Comment by jameslavery** [comment]
> https://stackoverflow.com/questions/50312937/skiasharp-tiff-support
> 
> I found a stackoverflow question for I think using this would optimize things
Thanks - that looks interesting, I'll take a look and see if I can understand it!

**Comment by jameslavery** [comment]
> You could always save mrasterfeature images byte array to the file system and load it from there instead of processing the tiff file

Yes that's what I was thinking, once it's processed (I'll probably look at optimising the processing too, if I can).
